CodeGeneration/Sources/generate-swift-syntax/LayoutNode+Extensions.swift

```diff
@@ -16,12 +16,65 @@ import SyntaxSupport
 import Utils
 
 extension LayoutNode {
-  func generateInitializerDeclHeader(useDeprecatedChildName: Bool = false) -> SyntaxNodeString {
+  /// Generates a memberwise SyntaxNode initializer `SyntaxNodeString`.
+  ///
+  /// - parameters:
+  ///  - rule: The ``NodeInitRule`` to use for generating the initializer. Applying a rule will make some children non-optional, and set default values for other children.
+  ///  - useDeprecatedChildName: Whether to use the deprecated child name for the initializer parameter.
+  func generateInitializerDeclHeader(for rule: NodeInitRule? = nil, useDeprecatedChildName: Bool = false) -> SyntaxNodeString {
     if children.isEmpty {
       return "public init()"
     }
 
-    func createFunctionParameterSyntax(for child: Child) -> FunctionParameterSyntax {
+    func childParameterName(for child: Child) -> TokenSyntax {
```

Comment:
This is monstrously large — extract these helper functions up one level, make them private?